src/core/lib/security/credentials/tls/grpc_tls_credentials_options.h

```diff
@@ -45,11 +45,19 @@ struct grpc_tls_key_materials_config
   void set_pem_root_certs(grpc_core::UniquePtr<char> pem_root_certs) {
     pem_root_certs_ = std::move(pem_root_certs);
   }
+  void set_pem_root_certs(const char* pem_root_certs) {
+    // make a copy of pem_root_certs.
+    grpc_core::UniquePtr<char> pem_root_ptr(gpr_strdup(pem_root_certs));
```

Comment:
any reason we don't use `std::string` instead of `unique_ptr<char>`?

Reply:
My guess: this is to make sure we only keep one copy of the certificate, since [the certificate string could take a large piece of memory](https://github.com/grpc/grpc/issues/17743). It might be inefficient if we use `std::string` and the caller is not handling it correctly, and causing a copy of the string. 
Besides root certs, the peer certs and private key are actually [also defined this way](https://github.com/grpc/grpc/blob/master/src/core/lib/security/security_connector/ssl_utils.h#L175-L176) .

@yihuazhang please correct me if I was wrong, or I misunderstood something. Thanks!

Reply:
+1 on avoiding unnecessary memory copy. When initially writing the TLS code, it was also not possible to use C++ standard library in C-core.